src/NinjaTools.FluentMockServer.API/Proxy/Visitors/RequestMatcherEvaluationVisitor.cs

```diff
@@ -11,85 +12,80 @@
 
-        public ComparasionVisitor(HttpContext httpContext)
+        /// <summary>
+        /// Initializes a new instance of <see cref="RequestMatcherEvaluationVisitor"/>.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/> of the current ASPNETCORE scope.></param>
+        public RequestMatcherEvaluationVisitor(HttpContext httpContext)
         {
+            _cts = new CancellationTokenSource();
             HttpContext = httpContext;
         }
 
-        /// <inheritdoc cref="IVisitor{TVisitable}" />
-        /// <remarks> ✅ Entrypoint ✅ </remarks>
-        public double Visit(RequestMatcher visitable, CancellationToken token = default)
+        private CancellationTokenSource _cts;
+        private HttpContext HttpContext { get; }
+        private HttpRequest HttpRequest => HttpContext.Request;
+
+        /// <inheritdoc />
+        public int Visit(RequestMatcher visitable)
         {
-            Score += Visit(visitable.Cookies);
-            Score += Visit(visitable.Headers);
-            Score += Visit(visitable.Query);
-            Score += Visit(visitable.Path);
-            Score += Visit(visitable.Method);
-            Score += Visit(visitable.BodyMatcher);
-            return Score;
+            try
+            {
+                _cts ??= new CancellationTokenSource();
+                var token = _cts.Token;
+
+                var score = 0;
+                score += Visit(visitable.Cookies, token);
+                score += Visit(visitable.Headers, token);
+                score += Visit(visitable.Query, token);
+                score += Visit(visitable.Path, token);
+                score += Visit(visitable.Method, token);
+                score += Visit(visitable.BodyMatcher, token);
+
+                token.ThrowIfCancellationRequested();
+                return score;
+            }
+            catch (OperationCanceledException canceled)
+            {
+                _cts = null;
+                return 0;
+            }
         }
 
-        /// <inheritdoc />
-        public double Visit(RequestBodyMatcher? visitable, CancellationToken token = default)
+        #region VisitMembers
+
+         /// <inheritdoc />
+        public int Visit(Cookies visitable, CancellationToken token = default)
         {
             token.ThrowIfCancellationRequested();
-            if (!visitable?.IsEnabled() ?? true)
+            if (!visitable?.Any() ?? true)
                 return Pass();
 
-            Guard.Against.Null(visitable, nameof(visitable) + nameof(RequestBodyMatcher));
-            if (HttpRequest.Body is null)
-                return Fail();
-
-            HttpRequest.EnableBuffering();
-            HttpRequest.Body.Seek(0, SeekOrigin.Begin);
-            using var reader = new StreamReader(HttpRequest.Body);
-            var httpBody = reader.ReadToEnd();
+            Guard.Against.NullOrEmpty(visitable.Cookie, nameof(visitable) + nameof(Cookies));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['visitable' is null on at least one execution path.](https://app.codacy.com/manual/git_36/NinjaTools.FluentMockServer/pullRequest?prid=4928294)